### DISCUSSION TITLE
Static weapons not spawning with any crew!

### DISCUSSION THREAD

**Discussion by darkshadowguard** [Vdauphin/HeartsAndMinds#1676] [Question and answer] [answered]
Would appreciate help with this, I have wasted many hours trying to fix this to no avail!

Using a mixture of mods, here is what I have, using a bunch of random units for the crewmen section in desperation:

btc_type_divers = ["O_diver_F"];
btc_type_crewmen = ["O_medic_F","O_Fighter_Pilot_F","O_Soldier_GL_F","O_crew_F"];
btc_type_boats = ["optre_catfish_ins_unarmed_f","optre_catfish_ins_mg_f"];
btc_type_motorized = ["OPTRE_M12_ins_APC","OPTRE_M12_FAV_ins","TCF_Tauri_unarmed_puma","OPTRE_m1015_mule_cover_ins"];
btc_type_motorized_armed = ["OPTRE_M12_TD_ins","OPTRE_M12_LRV_ins","OPTRE_M12R_AA_ins","OPTRE_M12A1_LRV_ins","OPTRE_M413_MGS_INS","OPTRE_M412_IFV_INS","OPTRE_M493_M37_Ins","OPTRE_M494_INS","VES_M19_URF_HMG","TCF_Tauri_armed_puma","TCF_Tauri_ferret"];
btc_type_mg = ["O_G_HMG_02_high_F","TCF_M48_Buzz_FRI","TCF_M92_BOLT_FRI","TCF_M247H_FRI","TCF_FRI_LAU","OPTRE_Static_FG75","OPTRE_Static_AA_Ins"];
btc_type_gl = ["O_G_HMG_02_high_F","TCF_M48_Buzz_FRI","TCF_M92_BOLT_FRI","TCF_M247H_FRI","TCF_FRI_LAU","OPTRE_Static_FG75","OPTRE_Static_AA_Ins"];

I have tried to change the crewmen listed in btc_type_crewmen, it never changes anything nor causes the assets to be crewed correctly, they still spawn empty. I have tried to add units by doing: " btc_type_crewmen = btc_type_crewmen + ["TCF_URA_Rifleman"];", it has not resulted in anything.

Attached is a screenshot version of my config, and at least one picture of the issue. Main faction being used is the OPTRE Insurrectionist faction, with the custom units from The Carver Findings as shown above.

Help would be deeply appreciated, I am at my wits end with this. I fixed this by accident before but that was literal years ago. What the hell is going wrong?

![107410_20241228110721_1](https://github.com/user-attachments/assets/88e1efad-4a76-4c85-8701-81b9c07e197c)

![107410_20241228110646_1](https://github.com/user-attachments/assets/1be9c496-1f29-4bb6-bdb2-eb1d3d7142d3)

![attempts](https://github.com/user-attachments/assets/bc7fdc64-c30c-48b6-94af-14903a85c0ab)

**Comment by Vdauphin** [comment, marked as answer]
Hello,

`btc_type_crewmen` is a string not an array (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L656C9-L656C25)
You should see an error in your .rpt about this

Cheers

**Reply by darkshadowguard** [reply]
Greetings

Thank you for the fast response,

I'll take a look a bit later and see if I can figure it out. Will get back to you if I'm having issues

**Reply by darkshadowguard** [reply]
I am confirming I was able to solve the problem with your assistance, thank you so much, it was driving me crazy.

For anyone who might stumble on this later, an example of a working config with the turrets being crewed correctly by a custom unit:

//Save class name to global variable
btc_enemy_side = _allclasse select 0;
btc_type_units = ["TCF_URA_Rifleman","TCF_URA_Rifleman_AT","TCF_URA_Marksman","TCF_URA_GL","TCF_URA_Engineer","TCF_URA_Medic","TCF_URA_Breacher","TCF_URA_AutoRifleman","TCF_URA_RTO","TCF_URA_Sniper","TCF_URA_TL","TCF_URA_SL","TCF_KOS_Rifleman_AA2"];
btc_type_divers = ["O_diver_F"];
btc_type_crewmen = "TCF_URA_Rifleman";
btc_type_boats = ["optre_catfish_ins_unarmed_f","optre_catfish_ins_mg_f"];
btc_type_motorized = ["OPTRE_M12_ins_APC","OPTRE_M12_FAV_ins","TCF_Tauri_unarmed_puma","OPTRE_m1015_mule_cover_ins"];
btc_type_motorized_armed = ["OPTRE_M12_TD_ins","OPTRE_M12_LRV_ins","OPTRE_M12R_AA_ins","OPTRE_M12A1_LRV_ins","OPTRE_M413_MGS_INS","OPTRE_M412_IFV_INS","OPTRE_M493_M37_Ins","OPTRE_M494_INS","VES_M19_URF_HMG","TCF_Tauri_armed_puma","TCF_Tauri_ferret"];
btc_type_mg = ["O_G_HMG_02_high_F","TCF_M48_Buzz_FRI","TCF_M92_BOLT_FRI","TCF_M247H_FRI","TCF_FRI_LAU","OPTRE_Static_FG75","OPTRE_Static_AA_Ins"];
btc_type_gl = ["O_G_HMG_02_high_F","TCF_M48_Buzz_FRI","TCF_M92_BOLT_FRI","TCF_M247H_FRI","TCF_FRI_LAU","OPTRE_Static_FG75","OPTRE_Static_AA_Ins"];

//Sometimes you need to remove units: - ["Blabla","moreBlabla"];
//Sometimes you need to add units: + ["Blabla","moreBlabla"];
switch (_p_en) do {
    /*case "Myfactionexemple" : {
        btc_type_units = btc_type_units - ["Blabla","moreBlabla"];
        btc_type_divers = btc_type_divers + ["Blabla","moreBlabla"];
        btc_type_crewmen = "TCF_URA_Rifleman";
        btc_type_boats = btc_type_boats;
        btc_type_motorized = btc_type_motorized;
        btc_type_mg = btc_type_mg;
        btc_type_gl = btc_type_gl;

**Reply by Vdauphin** [reply]
Have fun :)
